situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java

```diff
@@ -76,14 +78,25 @@ public void testPipelineEndToEnd() throws IOException, InterruptedException {
         builder.withUsername("admin");
         builder.withPassword("admin");
         final RestHighLevelClient restHighLevelClient = builder.build().createClient();
-        final SearchRequest searchRequest = new SearchRequest(INDEX_NAME);
-        searchRequest.source(
-                SearchSourceBuilder.searchSource().size(100)
+        // Wait for data to flow through pipeline and be indexed by ES
+        await().atMost(10, TimeUnit.SECONDS).untilAsserted(
+                () -> {
+                    refreshIndices(restHighLevelClient);
+                    final SearchRequest searchRequest = new SearchRequest(INDEX_NAME);
+                    searchRequest.source(
+                            SearchSourceBuilder.searchSource().size(100)
+                    );
+                    final SearchResponse searchResponse = restHighLevelClient.search(searchRequest, RequestOptions.DEFAULT);
```

Comment:
nit: we just need to do the rest high level request in here, not the assertions and parsing

Reply:
I think we do need put assertion in the body b/c from the failure I've seen:

```
java.lang.AssertionError: expected:<20> but was:<0>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at 
```

exception is produced by

```
Assert.assertEquals(expectedDocuments.size(), foundSources.size());
```

instead of `restHighLevelClient.search`. So we have to keep on sending search requests until the response get our ingested data back.